app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,12 @@
+class LikesController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    @like = post.like.new(author: current_user)
+
+    if @like.save
+      redirect_to user_post_path(current_user, post)
```

Comment:
- [ ] You did a great job working on this milestone! 💯 I spotted the bug you were speaking about with the likes:

If you try to like any `Post` written by the `current_user` no error happens. However, if you try to like any other user `Post`, you get the following error:

![image](https://github.com/lorenzoworx/velvetScribe/assets/77069011/454723da-98f4-4495-b69f-ee5ad6e56899)

Let's debug this:

First, if you look closely at your function, you're setting up the creation of a new `Like` on behalf of the currently logged-in user. According to your application logs, this part works fine:

![image](https://github.com/lorenzoworx/velvetScribe/assets/77069011/d8575c1d-8793-42ed-84dc-07acd1d4e829)

However, the problem arises when it successfully saves the post. You're trying to redirect to the `user_post_path` but you're not using the `Post` author's id, instead, you're using the `current_user` id:

![image](https://github.com/lorenzoworx/velvetScribe/assets/77069011/f01bce2a-d026-4a55-8486-ad4b631682b6)

 For example, if you're on the route: `users/5/posts/28` and you are the user with id `1`. When you like this post, your function is going to redirect you to `users/1/posts/28` But `Post` with id `28` doesn't belong to user `1` and thus, the error appears. 🤓 🔧